src/ast/passes/codegen_llvm.cpp

```diff
@@ -790,6 +790,59 @@ void CodegenLLVM::visit(Call &call)
     expr_ = buf;
     expr_deleter_ = [this, buf]() { b_.CreateLifetimeEnd(buf); };
   }
+  else if (call.func == "pton")
+  {
+    Expression *addr;
+    auto af_type = AF_INET;
+    int addr_size = 4;
+
+    if (call.vargs->size() == 2)
+    {
+      addr = call.vargs->at(1);
+      auto af_type_val = bpftrace_.get_int_literal(call.vargs->at(0));
+      if (af_type_val.has_value() && *af_type_val == AF_INET6)
+      {
+        af_type = AF_INET6;
+        addr_size = 16;
+      }
+    }
+    else
+    {
+      addr = call.vargs->at(0);
+    }
+
+    llvm::Type *array_t = ArrayType::get(b_.getInt8Ty(), addr_size);
+    AllocaInst *buf;
+    if (af_type == AF_INET6)
+    {
+      buf = b_.CreateAllocaBPF(array_t, "addr6");
+    }
+    else
+    {
+      buf = b_.CreateAllocaBPF(array_t, "addr4");
+    }
+
+    auto addr_str = bpftrace_.get_string_literal(addr);
+    char dst[addr_size];
+    Value *octet;
+
+    auto ret = inet_pton(af_type, addr_str.c_str(), &dst);
+    if (ret != 1)
+    {
+      LOG(FATAL) << "inet_pton() call returns " << ret;
```

Comment:
This should be a standard `ERROR` as it may occur quite normally when the user supplies an invalid address string. Ideally, it should be detected in the semantic analyser and reported along with the call location.